src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java

```diff
@@ -341,19 +341,18 @@ public static int getFuelValue(@Nonnull ItemStack itemStack)
         int fuelValue = 0;
         for (IFuelHandler handler : fuelHandlers)
         {
-            if(handler.matches(itemStack))
-                fuelValue = Math.max(fuelValue, handler.getBurnTime(itemStack));
+            if(isFuelHandled(itemStack))
+                fuelValue = Math.max(fuelValue, handler.getBurnTime(itemStack).getAsInt());
         }
         return fuelValue;
     }
 
     public static boolean isFuelHandled(@Nonnull ItemStack itemStack){
-        boolean handled = false;
         for(IFuelHandler handler : fuelHandlers) {
-            if(handler.matches(itemStack))
-                handled = true;
+            if(handler.getBurnTime(itemStack) !=null && handler.getBurnTime(itemStack).isPresent())
```

Comment:
That method is not declared nonnull so I don't suggest you to have the null check.